lib/models/BucketPolicy.js

```diff
@@ -0,0 +1,127 @@
+const assert = require('assert');
+
+const errors = require('../errors');
+const { validateResourcePolicy } = require('../policy/policyValidator');
+
+/**
+ * Format of json policy:
+ *  {
+ *      "Id": "Policy id",
+ *      "Version": "version date",
+ *      "Statement": [
+ *          {
+ *              "Sid": "Statement id",
+ *              "Effect": "Allow",
+ *              "Principal": "*",
+ *              "Action": "s3:*",
+ *              "Resource": "arn:aws:s3:::examplebucket/bucket2/object"
+ *          },
+ *          {
+ *              "Sid": "Statement id",
+ *              "Effect": "Deny",
+ *              "Principal": {
+ *                  "AWS": ["arn:aws:iam::<account_id>", "different_account_id"]
+ *              },
+ *              "Action": [ "s3:*" ],
+ *              "Resource": [
+ *                  "arn:aws:s3:::examplebucket", "arn:aws:s3:::otherbucket/*"],
+ *              "Condition": {
+ *                 "StringNotLike": {
+ *                      "aws:Referer": [
+ *                          "http://www.example.com/", "http://example.com/*"]
+ *                  }
+ *              }
+ *          }
+ *      ]
+ *  }
+ */
+
+class BucketPolicy {
+    /**
+     * Create a Bucket Policy instance
+     * @param {string} json - the json policy
+     * @return {object} - BucketPolicy instance
+     */
+    constructor(json) {
+        this._json = json;
+        this._policy = {};
+    }
+
+    /**
+     * Get the bucket policy
+     * @return {object} - the bucket policy or error
+     */
+    getBucketPolicy() {
+        const policy = this._getPolicyJSON();
+        return policy;
+    }
+
+    /**
+     * Get the bucket policy array
+     * @return {object} - contains error if policy validation fails
+     */
+    _getPolicyJSON() {
+        const policy = {};
+        if (!this._json || this._json === '') {
+            policy.error = errors.MalformedPolicy.customizeDescription(
+                'request json is empty or undefined');
+            return policy;
+        }
+        const validSchema = validateResourcePolicy(this._json);
+        if (validSchema.error) {
+            policy.error = validSchema.error;
+            return policy;
+        }
+        this._getStatementArray();
+        const valAcRes = this._validateActionResource();
+        if (valAcRes.error) {
+            policy.error = valAcRes.error;
+            return policy;
+        }
+
+        return this._policy;
+    }
+
+    _getStatementArray() {
```

Comment:
This seems more like a setter as it always returns undefined and sets the internal `this._policy`.